charts/meilisearch/templates/serviceMonitor.yaml

```diff
@@ -0,0 +1,42 @@
+{{- if.Values.serviceMonitor.enabled -}}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "meilisearch.fullname" . }}
+  labels:
+    {{- include "meilisearch.labels" . | nindent 4 }}
+    {{- with .Values.serviceMonitor.additionalLabels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  jobLabel: {{ .Release.Name }}
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace }}
+  selector:
+    matchLabels:
+      {{- include "meilisearch.selectorLabels" . | nindent 6 }}
+  endpoints:
+    - port: http
```

Comment:
Haven't tested this myself yet, but I was looking at adding a ServiceMonitor to my deployment, and I think you'd need something like this to support authentication, right?
```yaml
  endpoints:
    - bearerTokenSecret:
        name: meilisearch-master-key
        key: MEILI_MASTER_KEY
...
```